lesson18/src/js/main.js

```diff
@@ -0,0 +1,275 @@
+  const fragment = document.createDocumentFragment();
+  for(let i = 0; i < item; i++){
+    const pagenationBullet = createElementWithClassName("span", "bar");
+    pagenationBullet.style.width = "calc(width / item)";
+    pagenationBullet.setAttribute ("role", "button");
+    pagenationBullet.dataset.pagenationIndex = i + 1;
+    fragment.appendChild(pagenationBullet);
+
+    i === 0 && pagenationBullet.setAttribute("aria-current","true");
+
+  }
+  const slideArea = document.querySelector('.mainvisual__images__inner');
+  slideArea.appendChild(pagenationArea).appendChild(fragment);
+}
+
+function addNoimage(){
+  const slideImageArea = document.querySelector(".mainvisual__images__inner");
+  const sliderImage = createElementWithClassName("img", "slider__image");
+  sliderImage.src = './images/slide-image01.png'
+  sliderImage.width = 800;
+  sliderImage.height = 800;
+  sliderImage.style.opacity = 1;
+  slideImageArea.appendChild(sliderImage);
+}
+
+function switchPagination(currentIndex) {
+  const pagenationCurrent = document.getElementById("js-current");
+  pagenationCurrent.textContent = currentIndex;
+
+}
+
+function switchImg(target) {
+  const active = document.querySelector(".is-active");
+  active.classList.remove("is-active");
+  const changeSlide = document.querySelector(`[data-slide-index="${target}"]`);
+  changeSlide.classList.add("is-active");
+}
+
+function switcPagenationBullet(target){
+  const pagenationBullet = document.querySelector('[aria-current="true"');
+  pagenationBullet.setAttribute('aria-current', false);
+  const pageNations = document.querySelector(`[data-pagenation-index="${target}"]`);
+  pageNations.setAttribute("aria-current", true);
+
+}
+
+function clickPagenationBullet(){
+  const pagenationBullet = [...document.getElementsByClassName("bar")];
+  pagenationBullet.forEach((targetButton) => {
+    targetButton.addEventListener('click',(e) => {
+      const activePagenationBullet = document.querySelector('[aria-current="true"]');
+      const clickedPagenationBulletNo = e.currentTarget.getAttribute('data-pagenation-index');
+      activePagenationBullet.setAttribute('aria-current','false');
+      e.currentTarget.setAttribute('aria-current','true');
+      switchSlider(clickedPagenationBulletNo);
+      switchPagination(clickedPagenationBulletNo);
+      resetAutoPlaySlide();
+    })
+  })
+
+}
+
+function toggleButtonDisabled(index) {
+  const slideImages = [...document.getElementsByClassName("slider__item")];
+  const firstIndex = 1;
+  const lastIndex = slideImages.length;
+  const prevButton = document.getElementById("js-button_previous");
+  const nextButton = document.getElementById("js-button_next");
+
+  prevButton.disabled = Number(index) === firstIndex;
+  nextButton.disabled = Number(index) === lastIndex;
+}
+
+function renderSlidContents(slideImageData){
+  renderSlideImage(slideImageData);
+  renderPagenation(slideImageData.length);
+  renderPagenationBullet(slideImageData.length);
```

Comment:
same

Reply:
Thank you for your review.
Fixed all spelling mistakes.

[d137003](https://github.com/nyattsu72/hanson-frontend-cource/pull/26/commits/d137003ce0f2c77e2c22a737b1ef8ac77d843102)